packages/core/src/amazonqFeatureDev/session/session.ts

```diff
@@ -59,14 +64,51 @@ export class Session {
      * Preload any events that have to run before a chat message can be sent
      */
     async preloader(msg: string) {
-        if (!this.preloaderFinished) {
-            await this.setupConversation(msg)
-            this.preloaderFinished = true
-            this.messenger.sendAsyncEventProgress(this.tabID, true, undefined)
-            await this.proxyClient.sendFeatureDevTelemetryEvent(this.conversationId) // send the event only once per conversation.
+        const root = this.config.workspaceRoots[0]
+        const autoBuildProjectSetting = CodeWhispererSettings.instance.getAutoBuildFeatureProjects()
+        const hasDevfile = await checkForDevFile(root)
+        const isPromptedForAutoBuildFeature = Object.keys(autoBuildProjectSetting).includes(root)
+
+        if (hasDevfile && !isPromptedForAutoBuildFeature) {
```

Comment:
Can we extract the logic out of `preloader` function?

Reply:
as in, these 4 lines of code? sure

Reply:
actually, what are you suggesting here? I think this logic sits well here, do not need to extract the logic out, since it's part of pre-loading the conversation